### PR TITLE
fix(linter): ensure .cjs config file is handled correctly for generators #28214

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

When the workspace has `eslint.config.cjs` at the root, generation of new lint projects currently does not check for it correctly 

## Expected Behavior

Ensure utils continue to work as expected when `eslint.config.cjs` is detected at the workspaceRoot

## Related Issue(s)

Fixes #28214

### CODE REVIEW

File: packages/eslint/src/generators/utils/eslint-file.ts [hunk 1]
```diff
@@ -64,7 +64,11 @@ export function isEslintConfigSupported(tree: Tree, projectRoot = ''): boolean {
   if (!eslintFile) {
     return false;
   }
-  return eslintFile.endsWith('.json') || eslintFile.endsWith('.config.js');
+  return (
```
Comment (meeroslav) [inline packages/eslint/src/generators/utils/eslint-file.ts:67]: Should we also include `mjs` here? Use regex instead to match all three.

File: packages/eslint/src/generators/lint-project/lint-project.ts [hunk 2]
```diff
@@ -273,7 +273,8 @@ function createEsLintConfiguration(
     });
     const nodeList = createNodeList(importMap, nodes);
     const content = stringifyNodeList(nodeList);
-    tree.write(join(projectConfig.root, 'eslint.config.js'), content);
+    const ext = extendedRootConfig?.endsWith('.cjs') ? '.cjs' : '.js';
+    tree.write(join(projectConfig.root, `eslint.config${ext}`), content);
```
Comment (meeroslav) [inline packages/eslint/src/generators/lint-project/lint-project.ts:277]:
```suggestion
    const ext = extendedRootConfig?.split('.').pop() || 'js';
    tree.write(join(projectConfig.root, `eslint.config.${ext}`), content);
```

**Review by meeroslav — APPROVED** [review]

File: packages/eslint/src/generators/lint-project/lint-project.ts (hunk 2 above)
Comment (Coly010) [reply packages/eslint/src/generators/lint-project/lint-project.ts:277]: Let's support MJS in a separate PR

File: packages/eslint/src/generators/utils/eslint-file.ts (hunk 1 above)
Comment (Coly010) [reply packages/eslint/src/generators/utils/eslint-file.ts:67]: Let's support MJS in a separate PR

**Pull request merged**